### PR TITLE
Controller functions

### PR DESCRIPTION
**Pull request by speschellk**

### CODE REVIEW

**Comment by just1jp** [comment]
@BrennerSpear To save you some time, I skimmed this over. It looks good to me.

File: src/server/controller.js [hunk 1]
```diff
@@ -1,87 +1,189 @@
...
-      // res.send('controller signup.post');
+      // Check database for username
+      db.User.findAll({
+        where: { username: req.body.username }
+      })
+        .then(function(users) {
+          // If username is free
+          if (users.length === 0) {
+            // Hash password
+            bcrypt.hash(req.body.password, saltRounds, function(err, hash) {
```
Comment (BrennerSpear) [inline src/server/controller.js]: I think salt is used by default - we don't need to specify it

File: src/server/controller.js [hunk 2]
```diff
@@ -1,87 +1,189 @@
...
   upvote: {
     post: function(req, res) {
-      // increment 'votes' on post in posts table
-
-      // res.send('controller upvote.post');
+      db.Post.findOne({
+        where: { id: req.body.id },
+      })
+        .then(function(result) {
+          result.increment('vote_count');
```
Comment (BrennerSpear) [inline src/server/controller.js]: does 'increment' need a 'save' after it? I don't know myself, just wondering

File: src/server/controller.js (hunk 1 above)
Comment (speschellk) [inline src/server/controller.js]: True, will change to null and delete the saltRounds var

Comment (speschellk) [inline src/server/controller.js]: Actually, I can't tell why yet, but I get an execution error when I omit the salt. Keeping it in for now.

File: src/server/controller.js (hunk 2 above)
Comment (speschellk) [inline src/server/controller.js]: It doesn't seem to need a call to save. The vote count persists in the database after increment or decrement.

**Pull request merged**